src/workerd/api/actor-alarms-test.js

```diff
@@ -0,0 +1,67 @@
+import * as assert from 'node:assert'
+
+async function timeout(ms) {
+  await scheduler.wait(ms);
+  throw new Error(`timed out`);
+}
+
+export class DurableObjectExample {
+  constructor(state, env) {
+    this.state = state;
+  }
+
+  async waitForAlarm(scheduledTime, timeoutOverrideMs) {
+    console.log(`waiting for ${scheduledTime.valueOf()}`);
+    let self = this;
+    let prom = new Promise((resolve) => {
+      self.resolve = resolve;
+    });
+
+    let timeMs = scheduledTime.valueOf();
+    let timeoutMs = (timeMs - Date.now().valueOf()) + (timeoutOverrideMs ?? 4000);
+    try {
+      await Promise.race([prom, timeout(timeoutMs)]);
+      if (Date.now() < scheduledTime.valueOf()) {
+        throw new Error(`Date.now() is before scheduledTime! ${Date.now()} vs ${scheduledTime.valueOf()}`);
+      }
+    } catch (e) {
+      throw new Error(`error waiting for alarm at ${scheduledTime.valueOf()}: ${e}`);
+    }
+
+    let alarm = await this.state.storage.getAlarm();
+    if (alarm != null) {
+      throw new Error(`alarm time not cleared when handler ends. ${alarm}`);
+    }
```

Comment:
Suggestion: you can use `assert.ok(alarm == null, `alarm time not cleared when handler ends. ${alarm}`)` to simplify these checks.